cedar-policy/src/api.rs

```diff
@@ -2124,19 +1557,10 @@ impl EntityUid {
     pub fn from_json(json: serde_json::Value) -> Result<Self, impl miette::Diagnostic> {
```

Comment:
I left this in api.rs because it can return an error, and I was trying to maintain the hierarchy id.rs -> err.rs -> api.rs. I'm imagining that we'll want to update the return types here to use the types that will be defined (eventually) in err.rs.

Same answer for the comment on the `FromStr` impl below.

Reply:
I'm pretty sure Rust allows cycles in module imports (within the same crate), if that makes a difference :)